### ISSUE TITLE
pixel_art.effect errors due to illegal use of reserved word

### ISSUE THREAD

**Issue by flexiondotorg** [dspstanky/pixel-art#4]
Thanks for making this pixel-art filter! :smiley: 

I'm running OBS Studio 29.1.3 on Ubuntu, and adding the Pixel Art filter to a source does nothing. Inspecting the logs, these errors are presented:

```
error: Error compiling shader:
0:77(7): error: illegal use of reserved word `output'
0:77(7): error: syntax error, unexpected ERROR_TOK, expecting ',' or ';'


error: device_pixelshader_create (GL) failed
error: Pass (0) <> missing pixel shader!
error: Failed to create source 'Pixel Art'!
```

If I change all references of `output` to `result` (using `sed -i 's/output/result/g' pixel_art.effect`), compiling the shader now fails with different errors:

```
error: Error compiling shader:
0:45(2): error: initializer of type float cannot be assigned to variable of type int
0:46(2): error: initializer of type float cannot be assigned to variable of type int
0:50(2): error: C-style initialization requires the GL_ARB_shading_language_420pack extension
0:55(2): error: C-style initialization requires the GL_ARB_shading_language_420pack extension
0:62(2): error: C-style initialization requires the GL_ARB_shading_language_420pack extension
0:72(23): error: C-style initialization requires the GL_ARB_shading_language_420pack extension


error: device_pixelshader_create (GL) failed
error: Pass (0) <> missing pixel shader!
error: Failed to create source 'Pixel Art'!
```

**Comment by dspstanky** [comment]
Ah I see what I did wrong, needed to explicitly cast some float math to int, rather than relying on implicit cast, gonna do some more testing to make sure nothing else crops up, but should have a fixed release up later today, thank you!

**Comment by dspstanky** [comment]
ooh boy this problem is not as easy a fix as I hoped, the issue appears to be that when built for linux, OBS is treating an effect file as GLSL instead of HLSL like on windows. GLSL doesn't support the functionality i'm using to hardcode my bayer dithering arrays (or at least the version of GLSL that OBS is using doesn't). So I'll need to figure out a new way to implement those and change algorithm accordingly.

**Comment by dspstanky** [comment]
Fixed with release 0.0.2! However, performance might take a slight hit. Will be trying to optimize algorithm over coming days and will push another release when done. Wanted to at least get something functional out the door for linux users.

**Comment by flexiondotorg** [comment]
Thank you!
